modules/ballerina-http/src/main/ballerina/ballerina/net/http/request.bal

```diff
@@ -1,91 +1,237 @@
+@Param {value:"req: A request message"}
+@Param {value:"headerName: The header name"}
+@Param {value:"headerValue: The header value"}
 public function <Request req> setHeader (string headerName, string headerValue) {
-    if (req.headers == null) {
-        req.headers = {};
+    mime:Entity entity = req.getEntity(false);
+    if (entity.headers == null) {
+        entity.headers = {};
     }
-    HeaderValue[] header = [{value:headerValue}];
-    req.headers[headerName] = header;
+    mime:HeaderValue[] header = [{value:headerValue}];
+    entity.headers[headerName] = header;
 }
 
-@Description { value:"Removes a transport header from the request"}
-@Param { value:"req: A request message" }
-@Param { value:"key: The header name" }
+@Description {value:"Removes a transport header from the request"}
+@Param {value:"req: A request message"}
+@Param {value:"key: The header name"}
 public function <Request req> removeHeader (string key) {
-    if (req.headers == null) {
+    mime:Entity entity = req.getEntity(false);
+    if (entity.headers == null) {
         return;
     }
-    req.headers.remove(key);
+    entity.headers.remove(key);
 }
 
-@Description { value:"Removes all transport headers from the message"}
-@Param { value:"req: A request message" }
+@Description {value:"Removes all transport headers from the message"}
+@Param {value:"req: A request message"}
 public function <Request req> removeAllHeaders () {
-    req.headers = {};
+    mime:Entity entity = req.getEntity(false);
+    entity.headers = {};
 }
 
-@Description { value:"Checks whether the client expects a 100-continue response."}
-@Param { value:"req: A request struct" }
-@Return { value:"Returns true if the client expects a 100-continue response. If not, returns false." }
-public function <Request req> expects100Continue() (boolean) {
+@Description {value:"Checks whether the client expects a 100-continue response."}
+@Param {value:"req: A request struct"}
+@Return {value:"Returns true if the client expects a 100-continue response. If not, returns false."}
+public function <Request req> expects100Continue () (boolean) {
     var expectHeader = req.getHeader(HEADER_KEY_EXPECT);
     if (expectHeader != null && expectHeader.value == HEADER_VAL_100_CONTINUE) {
         return true;
     }
     return false;
 }
+
+@Description {value:"Gets the Content-Length header from the request"}
+@Param {value:"req: A request message"}
+@Return {value:"length of the message"}
+public function <Request request> getContentLength () (int) {
+    if (request.getHeader("Content-Length") != null) {
+        string strContentLength = request.getHeader("Content-Length").value;
+        var contentLength, conversionErr = <int>strContentLength;
+        if (conversionErr != null) {
+            throw conversionErr;
+        }
+        return contentLength;
+    } else {
+        error err = {msg:"No Content-Length header found!"};
+        throw err;
```

Comment:
+1

Reply:
Error was not returned to the end user, considering the usability. If we return the error, then every time the user wants to get the content length he has to declare '_' as well which would annoy the user to no end. I have changed the code to return -1 in case of no header and if a conversion error occured which is a scenario that is highly unlikely to happen. Please see the changes.